Jint/Native/Array/ArrayInstance.cs

```diff
@@ -462,6 +462,11 @@ private static bool IsArrayIndex(string p, out uint index)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static uint ParseArrayIndex(string p)
         {
+            if (p.Length == 0)
+            {
+                return uint.MaxValue;
+            }
+
             int d = p[0] - '0';
```

Comment:
I'm sorry to interfere, but why not use char.IsDigit(p[0]) here? I checked the performance of both approaches - they are identical.

Reply:
Well we need the actual value, say 5 to return and then when we detect length being more than 1 when start addition to that value.